src/utils/token_to_fiat_oracle.ts

```diff
@@ -0,0 +1,133 @@
+import type { FiatOracle } from './fiat_oracle';
+import type { CoinGeckoPriceRequestParams, CoinGeckoPriceResponseData } from './coingecko_types';
+import { TokenFiatPair } from './token_fiat_price';
+import type { FiatID, TokenID } from './fiat_oracle_types';
+
+const pollingIntervalMilliseconds = 1000 * 60 * 15; // 15 min
+
+function cachePairFilterFactory(fiat: FiatID, token: TokenID) {
+	return (p: TokenFiatPair) => p.fiat === fiat && p.token === token;
+}
+
+/*
+ * The CoinGecko wrapper does not support browser.
+ * Here it is the reported feature request for supporting
+ * the browser platform by just removing node dependencies.
+ *
+ * https://github.com/miscavage/CoinGecko-API/issues/19
+ *
+ * When the report is taken and fixed, please:
+ * 1. Import the CoinGecko wrapper.
+ * 2. Define a CoinGecko client as private readonly at the
+ *   constructor arguments. So we can test it.
+ * 3. On each call to the API, replace the fetch for the actual
+ *   API call.
+ *
+ * Examples
+ *
+ * 1. import CoinGecko from 'coingecko-api'
+ * 2. constructor(private readonly client = new CoinGecko()) {}
+ * 3. const response = await this.client.ping() // for the ping() function
+ *    const response = await this.client.simple.price(params) // for getting the price of a currency
+ */
+
+export class TokenToFiatOracle implements FiatOracle {
+	private cachedPrices: TokenFiatPair[] = [];
+	private cacheTimestamp = 0;
+	private syncPromise?: Promise<void>;
+
+	/**
+	 * @param {FiatID[]} fiats The currency IDs to quote arweave in
```

Comment:
Nit: `... to quote the tokens in`

Reply:
There: a29d13e2b27bbfc5d8c94b402fe1cc1cbb4013db